web-server/src/components/WorkflowSelector.tsx

```diff
@@ -0,0 +1,170 @@
+import {
+  Checkbox,
+  Chip,
+  FormControl,
+  InputLabel,
+  ListItemText,
+  MenuItem,
+  OutlinedInput,
+  Select,
+  CircularProgress
+} from '@mui/material';
+import { union } from 'ramda';
+import { FC, useCallback, useMemo } from 'react';
+
+import { handleApi } from '@/api-helpers/axios-api-instance';
+import { Line } from '@/components/Text';
+import { track } from '@/constants/events';
+import { CIProvider, Integration } from '@/constants/integrations';
+import { useAuth } from '@/hooks/useAuth';
+import { useEasyState, useBoolState } from '@/hooks/useEasyState';
+import {
+  BaseRepo,
+  RepoWorkflowResponse,
+  RepoWorkflow
+} from '@/types/resources';
+import { depFn } from '@/utils/fn';
+
+import { AsyncSelectOptions } from './AsyncSelect';
+import { FlexBox } from './FlexBox';
+import { useTeamCRUD } from './Teams/useTeamsConfig';
+
+export const DeploymentWorkflowSelector: FC<{ repo: BaseRepo }> = ({
+  repo
+}) => {
+  const { orgId } = useAuth();
+
+  const options = useEasyState<AsyncSelectOptions>([]);
+  const nextPageToken = useEasyState<string | null>('');
+  const loading = useBoolState();
+  const { updateWorkflowsForTeam } = useTeamCRUD();
+  const provider = Integration.GITHUB;
+
+  const selectedOptions = useMemo(
+    () =>
+      repo.repo_workflows?.map((val) => ({
+        label: val?.name,
+        value: val?.value
+      })) || [],
+    [repo.repo_workflows]
+  );
+  const loadWorkflows = useCallback(async () => {
+    if (!orgId) return;
+
+    track('VIEW_WORKFLOWS_FOR_REPO', { repo });
+
+    const workflowsResponse: RepoWorkflowResponse = await depFn(
+      loading.trackAsync,
+      () =>
+        handleApi<RepoWorkflow[]>(`/internal/${orgId}/integrations/workflows`, {
+          params: {
+            provider: provider,
+            org_name: repo.parent,
+            repo_name: repo.name,
+            repo_slug: repo.slug,
+            next_page_token: nextPageToken.value
+          }
+        })
+    );
+
+    const workflows = workflowsResponse.workflows;
+
+    nextPageToken.set(workflowsResponse.next_page_token);
+
+    const workflowOpts = workflows.map((w: RepoWorkflow) => ({
+      renderLabel: (
+        <FlexBox
+          fullWidth
+          justifyBetween
+          alignCenter
+          sx={{
+            fontSize: '12px'
+          }}
+        >
+          {w.name}
+          <Chip
+            label={
+              w.ci_provider === CIProvider.GITHUB_ACTIONS
+                ? 'Github Actions'
+                : 'Circle CI'
```

Comment:
Instead of this, we should have a provider label map with a default handling. Especially because CircleCI isn't supported here yet.

Reply:
removed this ternary condition as we don't have circle ci support yet